src/wp-includes/blocks.php

```diff
@@ -407,18 +408,49 @@ function register_block_type_from_metadata( $file_or_folder, $args = array() ) {
 	foreach ( $property_mappings as $key => $mapped_key ) {
 		if ( isset( $metadata[ $key ] ) ) {
 			$settings[ $mapped_key ] = $metadata[ $key ];
-			if ( $textdomain && isset( $i18n_schema->$key ) ) {
+			if ( $metadata_file_exists && $textdomain && isset( $i18n_schema->$key ) ) {
```

Comment:
If they register blocks via an array, they already have access to the WP translation functions like `__()` etc in PHP, so I don't think we'd need to account for anything... The `translate_settings_using_i18n_schema` function takes care of translating `block.json` data because there is no other way to provide translations for JSON files. If however, they don't use a JSON file at all, then there really is no reason to run that function since the user can already use PHP for their translations.
If they have a mix of block.json data and args, then the function will run properly 👍